cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -137,34 +137,28 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         walker.clearCallbacks()
         walker.registerHandler(this::handle)
 
-        // Resolve symbols in our translation units in the order depending on their import
-        // dependencies
-        for (tu in (Strategy::TRANSLATION_UNITS_LEAST_IMPORTS)(component)) {
-            log.debug("Resolving symbols of translation unit {}", tu.name)
-
-            // Gather all resolution EOG starters; and make sure they really do not have a
-            // predecessor, otherwise we might analyze a node multiple times
-            val nodes = tu.allEOGStarters.filter { it.prevEOGEdges.isEmpty() }
-
-            for (node in nodes) {
-                walker.iterate(node)
-            }
-        }
+        walker.iterate(eogStarter)
     }
 
     override fun cleanup() {
         templateList.clear()
     }
 
     /** This function caches all [TemplateDeclaration]s into [templateList]. */
-    private fun cacheTemplates(component: Component) {
+    private fun cacheTemplates(component: Component?) {
```

Comment:
I'm not really sure if we still need to get the templates for the `Component` or if it would be sufficient to fetch/cache those of the current `EOGStarter`.

Reply:
I hve no idea and I dont want to touch the template code :D So if this makes the tests green -> go for it.